exporter/collector/metricsexporter.go

```diff
@@ -69,8 +91,12 @@ func newGoogleCloudMetricsExporter(
 		cfg:    cfg,
 		client: client,
 		mapper: metricMapper{cfg},
+		mds:    make(chan *metricpb.MetricDescriptor),
```

Comment:
Do we want any buffer for this channel? As is, one slow CreateMetricDescriptor call will block `me.mds <- md`

Reply:
In my mind CMD is optimistic.  I want to give it the minimal amount of resource and I don't care if any specific call fails.  The current state of it, we basically *have* to include logic around it, but it's dubious whether we want it in the long run.

I'm happy having it on its own "thread" churning away slowly.

Reply:
To rephrase aaron's point, the current implementation isn't really better than having the CMD call serialized with the CreateTimeSeries call.  Either way, a CreateMetricDescriptor call can block the CreateTimeSeries calls.  If you want CMD to be optimistic, you could have a buffered channel, but drop CMD calls if the buffer fills up.  Otherwise, you probably don't need to bother with the extra goroutine.

Reply:
That makes sense, but without a buffer (or non-blocking write), L126 in pushMetrics() will block until the background goroutine reads from the channel

Reply:
Decided to go with a buffer of a few, and I do some pre-filtering of MDs before shoveling in the buffer.   I think what we have now matches requirements.

Reply:
LGTM